WordPress/gutenberg
# Mobile - Highlight text - Force format for some edge cases

- Gutenberg mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4471

## Description
This PR fixes some edge cases when applying a highlighted text color format although there are still a few limitations on the native side (Aztec) but for most cases now it should work as expected.

For cases where the caret is positioned at the end of the text, it'll apply the format manually so new entries of text will use the selected color.

For edge cases where we don't have support yet, it won't try to apply the format to avoid confusion seeing this format in the toolbar enabled but not showing in the actual test. An example of this edge case would be placing the caret in the middle of a word and toggling on the highlighted text format.

Selecting a text to apply a color format should work as expected.

**On Android**, changing the color with the caret positioned a the end of a word without a space in between it'll change the color of the entire word.

## How has this been tested?

- Open the app with metro running
- Test the highlight text formatting including:
  - Selecting a text and apply a color
  - Start writing with a color selected
  - Write some text without a color and then select a color and write more text
  - Try to apply a color in the middle of a word and **expect** to not see any color selected in the toolbar (we don't support this case)

## Screenshots 

Example 1 (iOS) | Example 2 (iOS)
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/149565364-08aea438-c77e-482a-b104-f1d5cf7e70d7.gif" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/149565376-b8c76bc2-4b08-428e-b1e1-8ec818a5ea90.gif" width="200" /></kbd>

Example 1 (Android) | Example 2 (Android)
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/149565432-337f6ee2-f54c-49c0-b952-12258158eb20.gif" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/149565439-f2d1c064-b4da-4c70-8157-7eb81863558b.gif" width="200" /></kbd>

## Types of changes
Bug fix

## Checklist:
